app/models/order.rb

```diff
@@ -118,6 +119,11 @@ def delete_orders_packages
     end
   end
 
+  def update_transition_and_reason(event, opts)
+    fire_state_event(event)
+    update(cancellation_reason_id: opts[:cancellation_reason_id], cancel_reason: opts[:cancel_reason]) if opts[:cancellation_reason_id]
```

Comment:
Given that the keys in `opts` are the same as the field names, you can just do `update(opts)` as you have sanitized the allowed options in the `params` method of the controller. An even safer way to do this would be...

```
def update_transition_and_reason(event, cancel_opts)
  fire_state_event(event)
  opts = opts.select{|k,v| ['cancellation_reason_id', 'cancel_reason'].include?(k) }
  update(opts)
end
```